File: Stack.java
class Node{
  int data;
  Node next;

  public Node(int key){
    this.data = key;
    this.next = null;
  }
}

public class Stack{
  public static Node push(int key, Node head){
    Node newNode = new Node(key);
    if (head == null) return newNode;

    newNode.next = head;
    head = newNode;

    return head;
  }

  public static void traverse(Node head){
    if(head == null) return;

    do{
      System.out.print(head.data+" ");
      head = head.next;
    }while(head != null);
  }

  public static Node pop(Node head){
    if(head == null) return null;
    head = head.next;
    return head;
  }
  public static void main(String[] args){
    Node head = null;
    head = push(100,head);
    head = push(200,head);
    head = push(300,head);
    head = push(900,head);
    traverse(head);
    System.out.println("\n");
    head = pop(head);
    traverse(head);
    System.out.println("\n");
    head = pop(head);
    traverse(head);
    System.out.println("\n");
    head = pop(head);
    traverse(head);
    System.out.println("\n");
    head = pop(head);
    traverse(head);
    System.out.println("\n");
  }
}

File: Stack.c
// C Program for Stack operations


#include<stdio.h>
#include<stdlib.h>
struct Node{
	struct Node *next;
	int data;
};

void StackInsert(struct Node *node,int data);
void traverse(struct Node *head);

int main(){
	struct Node head;
	StackInsert(*head,10);
	
	return 0;
}

void StackInsert(struct Node *node, int data){
	struct Node *temp = NULL;
	temp->next = NULL;
	temp->data = data;

	if(node == NULL){
		node = temp;
	}
	else{
		node->next = temp;
	}
	free(temp);
}

void traverse(struct Node *head){
	if(head == NULL || head->next == NULL){
		printf("End of Stack");
		return;
	}
	printf("Data -> %d", head->data);
	traverse(head->next);
}

